### PR TITLE
Add views

### PR DESCRIPTION
**Pull request by wacoo**
- Add layout
- Add partials
- Add views for user and post controllers focussing on index and show actions
- Fix linter errors

### CODE REVIEW

File: app/views/users/_user_info.html.erb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+            <div class="post">
+            <h2>Post: <%= post.title %></h2>
+            <p><%= post.text %></p>
+            </div>
+            <% counter += 1 %>
+        <% else %>
+            <% break %>
+        <% end %>
+    <% end %>
+    <button id="see-all">See all posts</button>
```
Comment (franclobo) [inline app/views/users/_user_info.html.erb]: - This button should link to the post#index page. Please 🙏🏽 add the function to the button to be able to see all posts.

File: app/views/users/_blogger.html.erb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+
+<div>
+    <div class = "photo">
+        <%= image_tag(user.photo) %>
+    </div>
+    <div class="user">
+        <h2 class="name"><%= user.name %></h2>
```
Comment (franclobo) [inline app/views/users/_blogger.html.erb]: - [ ] The user.name should link to the user#show page. Please 🙏🏽 add the function to the name to be able to see the user information.

File: app/views/users/_user_info.html.erb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    <% user.posts.each_with_index do |post, idx| %>
+        <% if counter < 2 %>
+            <div class="post">
+            <h2>Post: <%= post.title %></h2>
+            <p><%= post.text %></p>
+            </div>
+            <% counter += 1 %>
+        <% else %>
+            <% break %>
+        <% end %>
```
Comment (franclobo) [inline app/views/users/_user_info.html.erb]: - The user#show page should show the last three posts, it is showing just two. Please 🙏🏽 use the method `recent_three_posts` from your model to be able to see the last three posts.

image

File: app/views/posts/_post.html.erb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<div class="user-info-container">
+    <div class="post">
+        <h2>Post: <%= post.title %></h2>
```
Comment (franclobo) [inline app/views/posts/_post.html.erb]: - [ ] The post title should be a link to post#show page. Please 🙏🏽 add this function to the post title to be able to see all comments.

**Review by franclobo — COMMENTED** [review]
Hi @wacoo, 

# Required Changes ♻️

so-close-jim-halpert

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- Good use of partials 👏🏽 
- Correct Git flow ✔️ 
- Descriptive PR ✔️ 
- Good UI ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Kindly update your root route to users#index for a better UX.
- **[OPTIONAL]** Would you like to create users, posts, and comments automatically? We could add gem 'faker' in the Gemfile and add the following code in `config/db/seeds.rb` 👇🏽 

```ruby
require 'faker'

# Create 10 users
10.times do
  User.create(
    name: Faker::Name.name,
    photo: Faker::Avatar.image,
    bio: Faker::Lorem.paragraph(sentence_count: 2),
    posts_counter: 0
  )
end

# Create 100 posts
100.times do
  Post.create(
    author_id: rand(1..10),
    title: Faker::Lorem.sentence(word_count: 3),
    text: Faker::Lorem.paragraph(sentence_count: 3),
    comments_counter: 0,
    likes_counter: 0
  )
end

# Create 1000 comments
1000.times do
  Comment.create(
    user_id: rand(1..10),
    post_id: rand(1..100),
    text: Faker::Lorem.paragraph(sentence_count: 2)
  )
end
```
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/_user_info.html.erb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    </div>
+    <div class="display2 posts-relaxed">
+        <% user.recent_three_posts.each do |post, idx| %>
+            <div class="post">
+                <h2><%= link_to "Post: #{post.title}", user_post_path(post.author, post) %></h2>
+                <p><%= post.text %></p>
+            </div>
+        <% end %>    
+    </div>
+    <button id="all">See all posts</button>
```
Comment (nahnahdev) [inline app/views/users/_user_info.html.erb:14]: - [ ] Great job so far @wacoo There is just one change I will suggest you implement and you'll be on your way to the next Milestone :wink: 
- [ ] Great job making sure the `See all Posts` button shows the hidden posts. But from the given wireframe and the project requirements, just as the previous reviewer mentioned you need to ensure that the `See all Posts` button opens the `Posts Index Page` and shows all the posts for a given user. So kindly work on that :+1:
- Your Implementation

[Screencast from 11-10-2023 12:17:44 PM.webm](https://github.com/wacoo/blog-front-ruby-rails/assets/102699958/d7b2d71f-3c47-4709-b7c6-a963505e04bf)

- The Project Requirements

image

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @wacoo  , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which haven't been addressed yet.


 ## Highlights :fire: 
- LInters well setup and passing :heavy_check_mark: 
- Most requested changes implemented :heavy_check_mark: 
- App starts without any errors :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @NahnahAJ in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by divinecharlotte — APPROVED** [review]
# Hello @wacoo      😃 🥳 🎉 🎈 

Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say other than... it's time to merge it 👍🏻 ✅ 
### To Highlight 💯

- Good implementation of the required changes✔️
- Descriptive pr✔️






## Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**